Sources/WhisperKit/Core/WhisperKit.swift

```diff
@@ -530,12 +584,74 @@ open class WhisperKit {
         return result.first
     }
 
+    /// Main entry point for transcribing audio
+    /// - Parameters:
+    ///   - audioArray: Array of 16khz raw float audio samples
+    ///   - decodeOptions: Options for how to transcribe audio. Including a chunking strategy and the number of concurrent workers will paralleize this task.
+    ///   - callback: Optional callback to receive updates during the transcription process.
+    /// - Returns: An array of sorted `TranscriptionResult`.
+    /// - Throws: An error if the transcription fails.
     public func transcribe(
         audioArray: [Float],
         decodeOptions: DecodingOptions? = nil,
         callback: TranscriptionCallback = nil
     ) async throws -> [TranscriptionResult] {
-        if self.modelState != .loaded {
+        var transcribeResults = [TranscriptionResult]()
+
+        // Determine if the audio array requires chunking
+        if audioArray.count > WhisperKit.windowSamples, let chunkingStrategy = decodeOptions?.chunkingStrategy {
+            // We have some audio that will require multiple windows and a strategy to chunk them
+            switch chunkingStrategy {
+                case .vad:
+                    let chunker = VADAudioChunker()
+                    let audioChunks = try await chunker.chunkAll(
+                        audioArray: audioArray,
+                        maxChunkLength: WhisperKit.windowSamples,
+                        decodeOptions: decodeOptions
+                    )
+
+                    // Send chunked samples to transcribe (note: this is recursive)
+                    let chunkedResults: [Result<[TranscriptionResult], Swift.Error>] = await transcribe(
+                        audioArrays: audioChunks,
+                        decodeOptions: decodeOptions,
+                        callback: callback
+                    )
+
+                    transcribeResults = try chunkedResults.flatMap { try $0.get() }
+                @unknown default:
+                    break
+            }
+        }
+
+        // Audio is short enough to transcribe in a single window
+        if transcribeResults.isEmpty {
+            transcribeResults = try await runTranscribeTask(
+                audioArray: audioArray,
+                decodeOptions: decodeOptions,
+                callback: callback
+            )
+        }
```

Comment:
maybe it can be simplified to 

```swift
// Determine if the audio array requires chunking
let requiresChunking = audioArray.count > WhisperKit.windowSamples
switch (requiresChunking, decodeOptions?.chunkingStrategy) {
case (true, .vad):
    // We have some audio that will require multiple windows and a strategy to chunk them
    let chunker = VADAudioChunker()
    let audioChunks = try await chunker.chunkAll(
        audioArray: audioArray,
        maxChunkLength: WhisperKit.windowSamples,
        decodeOptions: decodeOptions
    )
    // Send chunked samples to transcribe (note: this is recursive)
    let chunkedResults: [Result<[TranscriptionResult], Swift.Error>] = await transcribe(
        audioArrays: audioChunks,
        decodeOptions: decodeOptions,
        callback: callback
    )
    transcribeResults = try chunkedResults.flatMap { try $0.get() }
default:
    // Audio is short enough to transcribe in a single window or no chunking strategy is provided
    transcribeResults = try await runTranscribeTask(
        audioArray: audioArray,
        decodeOptions: decodeOptions,
        callback: callback
    )
}
```

Reply:
Very nice ✨ was trying to come up with a cleaner way and forgot you could do this 😂